### DISCUSSION TITLE
Use own fine tuning model for completition.

### DISCUSSION THREAD

**Discussion by Majox21** [openai-php/client#219] [Q&A]
Hello,
I have done my first own model with your code, I tried to use it, but I do not know, how, I tried this:
```
$response = $client->completions()->create([
    'model' => 'ft:gpt-3.5-turbo-0613:personal::863ZS9QM',
    'prompt' => 'How long takes shipping?',
    'max_tokens' => 6,
    'temperature' => 0
]);

$response->id; // 'cmpl-uqkvlQyYK7bGYrRHQ0eXlWi7'
$response->object; // 'text_completion'
$response->created; // 1589478378
$response->model; // 'gpt-3.5-turbo-instruct'

foreach ($response->choices as $result) {
    $result->text; // '\n\nThis is a test'
    $result->index; // 0
    $result->logprobs; // null
    $result->finishReason; // 'length' or null
}

$response->usage->promptTokens; // 5,
$response->usage->completionTokens; // 6,
$response->usage->totalTokens; // 11

$response->toArray(); // ['id' => 'cmpl-uqkvlQyYK7bGYrRHQ0eXlWi7', ...]
echo '<pre>';
print_r($response);
```
**But I received an error:**

_Fatal error: Uncaught OpenAI\Exceptions\ErrorException: This is a chat model and not supported in the v1/completions endpoint. Did you mean to use v1/chat/completions?_

Could you please help me, how to use my fine tuned model with your client?

Thank you very much.

**Comment by pb30** [comment]
Use `$client->chat()->create` instead of `$client->completions()->create`

You'll also need to use the correct parameters for a `chat` call, like the `messages` array instead of `prompt`

**Reply by Majox21** [reply]
Thank you for a quick reply, so the right code will be:
```
$response = $client->chat()->create([
    'model' => 'ft:gpt-3.5-turbo-0613:personal::863ZS9SM',
    'messages' => [
        ['role' => 'user', 'content' => 'How long takes shipping?']
    ],
    'max_tokens' => 6,
    'temperature' => 0
]);
```

Is that right? Or I missed something?

**Reply by pb30** [reply]
Looks about right, but that `max_tokens` is pretty low, your response may be cut-off

**Reply by Majox21** [reply]
Yes, it is, I will try with more tokens. But I am a little surprised, that  for specific trained question and answer it outputs generative content instead of answer I provided at dataset. I have small dataset maybe 10 questions and answers and maybe 2 of them worked as I wish before. Temperature I set to 0. So I think I should make some more variations of questions. Or rather to use other model?, to have more strict answers based on dataset?

**Reply by pb30** [reply]
Generally fine-tuning requires hundreds of training examples to work properly. It's also not a silver bullet, if you are trying to teach the model new information using embeddings is a better route. You can read more in the official [fine tuning guide](https://platform.openai.com/docs/guides/fine-tuning) or try posting on the OpenAI forums with details on what you are trying to do and get some recommendations on the best route.

**Reply by Majox21** [reply]
Thank you for answer, I have checked that Embeddings, they output many vectors, so if I understood it right, I can work them and compare more strings etc. For live support for eshop do you think is better to work and play with embeddings or rather to make better dataset? I think that make better dataset will be better way and also faster, but that is only my option, I think that working with embeddings could be harder and slower, but maybe I have misunderstood something.
